tests/test_class.cpp

```diff
@@ -324,7 +324,7 @@ TEST_SUBMODULE(class_, m) {
     public:
     // Removing this destructor results in linking errors with the Intel compiler
     // (in Debug builds only, tested with icpc (ICC) 2021.1 Beta 20200827)
-    ~PublicistB() override {};
+    ~PublicistB() override {}; // NOLINT
```

Comment:
I added the same change in [my branch](https://github.com/tobiasleibner/pybind11/tree/topic-icc). That branch also contains some more workarounds to fix https://github.com/pybind/pybind11/issues/2707.

Reply:
Once we get this in could you PR the fixes needed? We could add a matrix with C++17 in that PR.

Reply:
> Once we get this in could you PR the fixes needed? We could add a matrix with C++17 in that PR.

@mkuron created a follow-up PR which also adds a C++17 CI run. We probably should merge #2573 first, though. It seems as if there is one ``// NOLINT`` missing and one Python test failing here. Any ideas about the Python test?